sparse_strips/vello_hybrid/examples/text.rs

```diff
@@ -0,0 +1,260 @@
+                            label: Some("Vello Render to Surface pass"),
+                        });
+                {
+                    let mut pass = encoder.begin_render_pass(&RenderPassDescriptor {
+                        label: Some("Render to Texture Pass"),
+                        color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                            view: &texture_view,
+                            resolve_target: None,
+                            ops: wgpu::Operations {
+                                load: wgpu::LoadOp::Clear(wgpu::Color::BLACK),
+                                store: wgpu::StoreOp::Store,
+                            },
+                        })],
+                        depth_stencil_attachment: None,
+                        occlusion_query_set: None,
+                        timestamp_writes: None,
+                    });
+                    self.renderers[surface.dev_id].as_mut().unwrap().render(
+                        &self.scene,
+                        &mut pass,
+                        &render_params,
+                    );
+                }
+
+                device_handle.queue.submit([encoder.finish()]);
+                surface_texture.present();
+
+                device_handle.device.poll(wgpu::Maintain::Poll);
+            }
+            _ => {}
+        }
+    }
+}
+
+fn draw_text(
+    ctx: &mut Scene,
+    text: String,
+    font_cx: &mut FontContext,
+    layout_cx: &mut LayoutContext<ColorBrush>,
+) {
+    let mut builder = layout_cx.ranged_builder(font_cx, &text, 1.0);
+    builder.push_default(FontFamily::parse("Roboto").unwrap());
+    builder.push_default(StyleProperty::LineHeight(1.3));
+    builder.push_default(StyleProperty::FontSize(32.0));
+
+    let mut layout: Layout<ColorBrush> = builder.build(&text);
+    let max_advance = Some(400.0);
+    layout.break_all_lines(max_advance);
+    layout.align(max_advance, Alignment::Middle, AlignmentOptions::default());
+
+    for line in layout.lines() {
+        for item in line.items() {
+            if let PositionedLayoutItem::GlyphRun(glyph_run) = item {
+                render_glyph_run(ctx, &glyph_run, 30);
+            }
+        }
+    }
+}
+
+fn render_glyph_run(ctx: &mut Scene, glyph_run: &GlyphRun<'_, ColorBrush>, padding: u32) {
+    let mut run_x = glyph_run.offset();
+    let run_y = glyph_run.baseline();
+    let glyphs = glyph_run.glyphs().map(|glyph| {
+        let glyph_x = run_x + glyph.x + padding as f32;
+        let glyph_y = run_y - glyph.y + padding as f32;
+        run_x += glyph.advance;
+
+        Glyph {
+            id: glyph.id as u32,
+            x: glyph_x,
+            y: glyph_y,
+        }
+    });
+
+    let run = glyph_run.run();
+    let font = run.font();
+    let font_size = run.font_size();
+    let normalized_coords = run
+        .normalized_coords()
+        .iter()
+        .map(|coord| NormalizedCoord::from_bits(*coord))
```

Comment:
This is where we'd use `bytemuck::cast_slice`, I think.

Reply:
https://github.com/linebender/vello/pull/883/commits/72ac1457a9598ef95144eea09747b3fc73b3e9db